tests/functional/backend/corpora/test_upload_sfn.py

```diff
@@ -0,0 +1,54 @@
+import json
+import os
+import unittest
+
+import boto3
+
+from backend.corpora.common.upload_sfn import start_upload_sfn
+
+if os.getenv("DEPLOYMENT_STAGE") == "test":
+    step_definition = json.loads(
+        """{
+      "Comment": "A Hello World example demonstrating various state types of the Amazon States Language",
+      "StartAt": "Pass",
+      "States": {
+        "Pass": {
+          "Type": "Pass",
+          "End": true
+        }
+      }
+    }
+    """
+    )
+    client_sfn = boto3.client("stepfunctions", endpoint_url=os.getenv("BOTO_ENDPOINT_URL"))
+    client_ssm = boto3.client("secretsmanager", endpoint_url=os.getenv("BOTO_ENDPOINT_URL"))
+    secret_id = "corpora/backend/test/config"
+    old_secret = client_ssm.get_secret_value(SecretId=secret_id)["SecretString"]
+
+
+    def setUpModule():
+        response = client_sfn.create_state_machine(
+            name="fake_step",
+            definition=json.dumps(step_definition),
+            roleArn="arn:aws:iam::123456789012:role/corpora-dataset-uploader-test-sfn-service",
```

Comment:
Does it work with this Arn? How?

Reply:
When the environment is test, it works with the test framework created by central infra using localstack. The docker containers must be running in the background for it to work. The fake role isn't used on any of the deployed environments. 

This test will not work in github since the containers aren't used. Central infra is working on that. The tests will work on all other deployments if the step function exists.